modules/core/src/main/scala/org/aossie/agora/model/Ballot.scala

```diff
@@ -45,7 +45,7 @@ case class RankBallot(val ranks: List[(Candidate, Int)], override val id: Int, w
 
 }
 
-case class ApprovalBallot(val approvals: Set[Candidate], override val id: Int, w: Rational)
+class ApprovalBallot(val approvals: Set[Candidate], override val id: Int, w: Rational)
```

Comment:
Follow same recommendation as stated for PrefenceBallot here. Add type parameter for Candidate say C and revert to case class.